manifests/repo/apt.pp

```diff
@@ -31,40 +31,32 @@
 # Copyright 2015 RL Solutions, unless otherwise noted
 #
 class gluster::repo::apt (
-  $version         = $::gluster::params::version,
-  $release         = $::gluster::params::release,
-  $priority        = $::gluster::params::repo_priority,
+  $version  = $::gluster::params::version,
+  $release  = $::gluster::params::release,
+  $priority = $::gluster::params::repo_priority,
 ) {
   include '::apt'
 
-  ensure_packages(['apt-transport-https'], {'ensure' => 'present'})
-
   # Key has changed since 3.9
   $repo_key_name = $release ? {
-    '3.8'                                     => 'A4703C37D3F4DE7F1819E980FE79BB52D5DC52DC',
-    '3.10'                                    => 'C784DD0FD61E38B8B1F65E10DAD761554A72C1DF',
-    '3.11'                                    => 'DE82F0BACC4DB70DBEF95CA65EC2255642304A6E',
-    '3.12'                                    => '8B7C364430B66F0B084C0B0C55339A4C6A7BD8D4',
-    default                                   => '849512C2CA648EF425048F55C883F50CB2289A17',
+    '3.10'  => 'C784DD0FD61E38B8B1F65E10DAD761554A72C1DF',
+    '3.11'  => 'DE82F0BACC4DB70DBEF95CA65EC2255642304A6E',
+    '3.12'  => '8B7C364430B66F0B084C0B0C55339A4C6A7BD8D4',
+    '3.13'  => '9B5AE8E6FD2581F293104ACC38675E5F30F779AF',
+    default => '9B5AE8E6FD2581F293104ACC38675E5F30F779AF',
   }
 
-  # Key location is different for some releases
-  $repo_key_source = $release ? {
-    '3.8'   => "https://download.gluster.org/pub/gluster/glusterfs/${release}/LATEST/rsa.pub",
-    default => "https://download.gluster.org/pub/gluster/glusterfs/${release}/rsa.pub",
-  }
+  $repo_key_source = "https://download.gluster.org/pub/gluster/glusterfs/${release}/rsa.pub"
 
   # basic sanity check
   if $version == 'LATEST' {
     $repo_ver = $version
+  } elsif $version =~ /^\d\.\d+$/ {
```

Comment:
would it make sense to convert this into a datatype?

Reply:
Not that much because AFAIK you can't really extract variables from a data type like we do in the case below. You could avoid the else case at the end, but an if/elsif/elsif tree without an else feels wrong to me.